readthedocs/profiles/views.py

```diff
@@ -75,11 +76,13 @@ class AccountDelete(PrivateViewMixin, SuccessMessageMixin, FormView):
     success_message = _('You have successfully deleted your account')
 
     def get_object(self):
-        return self.request.user
+        return User.objects.get(pk=self.request.user.pk)
```

Comment:
update_change_reason needs a proper instance, self.request.user was returning a LazyObject.

Reply:
Hrm -- seems like this is causing more issues than I expected. I think we should probably add exceptions around the `update_change_reason` instead of changing the code in these ways :/

Reply:
But then we won't have the correct change reason, I prefer having tests for these cases (we do have tests for deleting users). We should always pass a proper instance of an object to update_change_reason.

Reply:
Tests are great, but we definitely need exception handling given that it's already caused issues in testing. There will definitely be more edge cases in production, and a 500 is way worse than missing `change_reason`.